### PR TITLE
Use gometalinter config instead of makefile

### LINKED ISSUE #559: Rewrite lint.mk to a configuration file
# Problem

We store linter ([gometalinter](https://github.com/alecthomas/gometalinter)) configuration directly in a Makefile which is hard to find and there are a lot of duplications and inconsistency.

# Implementation

Rewrite `lint.mk` into `.gometalinter.json` as described [here](https://github.com/alecthomas/gometalinter#configuration-file).

# Acceptance Criteria

1. The same linters check the same packages,
2. Linter check passes in CI.

# Notes

It may happen that some options are not available in a configuration file. The first step should be carrying out research if `.gometalinter.json` has all features that we use via command line flags. If not, we should discuss how to bypass that.

### PR DESCRIPTION
**Pull request by dnephin**
Replace most of `lint.mk` with a `.gometalinter.json` to remove duplication

Closes #559

### CODE REVIEW

File: .gometalinter.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "Exclude": [
+        ".*_mock.go",
+        "geth/jail/doc.go"
+    ],
+    "Skip": ["helpers", "static", "testing", "e2e"],
```
Comment (adambabik) [inline .gometalinter.json]: `testing` and `e2e` should be included.

File: .gometalinter.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "Vendor": true,
+    "Enable": [
```
Comment (adambabik) [inline .gometalinter.json]: For all curious reviewers, `unused`, `staticcheck`, `gosimple` are part of `megacheck` so they are included.

File: .gometalinter.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (adambabik) [inline .gometalinter.json:1]: Can we also enable `--warn-unmatched-nolint`?

File: static/tools/mk/lint.mk [hunk 4]
```diff
@@ -1,64 +1,7 @@
...
-lint-varcheck:
-	@echo "lint-varcheck"
-	@gometalinter $(LINT_EXCLUDE) --disable-all --enable=varcheck --deadline=45s  $(LINT_FOLDERS)
-lint-structcheck:
-	@echo "lint-structcheck"
-	@gometalinter $(LINT_EXCLUDE) --disable-all --enable=structcheck --deadline=45s  $(LINT_FOLDERS)
-lint-gosimple:
-	@echo "lint-gosimple"
-	@gometalinter $(LINT_EXCLUDE) --disable-all --enable=gosimple --deadline=45s  $(LINT_FOLDERS)
+lint:
```
Comment (adambabik) [inline static/tools/mk/lint.mk]: Could you also move `lint-install` and `lint` targets to the main `Makefile` and delete `static/tools/mk/lint.mk` completely?

**Review by adambabik — CHANGES_REQUESTED** [review]
Thanks for the PR!

Unfortunately, it looks like our previous configuration was poor and many files were not linted (errors visible in Travis) :/ Before we merge this PR to develop we need to fix it. There are two options:
1. Fix code errors in this PR (but it was not within the scope of the issue),
2. Change base branch of this PR to a temporary one, merge this PR into that temporary branch and create another PR to fix the code.

Please let me know which one is the best for you.

**Comment by dnephin** [comment]
Thanks, I'll make these changes in the next couple of days

**Comment by dnephin** [comment]
I fixed the first set of lint errors, however adding `WarnUnmatchedDirective` adds another 60 errors from `nolint`.

**Comment by adambabik** [comment]
@dnephin let's skip `WarnUnmatchedDirective` for now then. I will create a follow-up issue.

**Comment by dnephin** [comment]
linting passed, e2e failed with a panic

**Review by dshulyak — APPROVED** [review]
thanks, that is known error, I restarted  a job

**Review by adambabik — APPROVED** [review]

**Comment by adambabik** [comment]
Thanks a lot @dnephin! It's so much better now.

**Pull request merged**